Rubberduck.Parsing/Symbols/ModuleDeclaration.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Rubberduck.Parsing.Annotations;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.Symbols
+{
+    public abstract class ModuleDeclaration : Declaration
+    {
+        protected ModuleDeclaration(
+            QualifiedMemberName qualifiedName,
+            Declaration projectDeclaration,
+            string name,
+            DeclarationType declarationType,
+            bool isUserDefined,
+            IEnumerable<IAnnotation> annotations,
+            Attributes attributes,
+            bool isWithEvents = false)
+            : base(
+                qualifiedName,
+                projectDeclaration,
+                projectDeclaration,
+                name,
+                null,
+                false,
+                isWithEvents,
+                Accessibility.Public,
+                declarationType,
+                null,
+                null,
+                Selection.Home,
+                false,
+                null,
+                isUserDefined,
+                annotations,
+                attributes)
+        { }
+
+        private readonly List<Declaration> _members = new List<Declaration>();
+        public IReadOnlyList<Declaration> Members => _members;
+
+        public void AddMember(Declaration member)
```

Comment:
shouldn't this be `internal`? We don't really want "external" consumers of Rubberduck.Parsing to mess with the state we parsed and resolved, except through rewriting and reparsing

Reply:
That one's on me.  Fixed.